source/slang/lower-to-ir.cpp

```diff
@@ -2781,6 +2781,30 @@ struct DeclLoweringVisitor : DeclVisitor<DeclLoweringVisitor, LoweredValInfo>
         return LoweredValInfo();
     }
 
+    void walkInheritanceHierarchyAndCreateWitnessTableCopies(IRWitnessTable* witnessTable, Type* subType, InheritanceDecl* inheritanceDecl)
+    {
+        auto baseDeclRef = inheritanceDecl->base.type.As<DeclRefType>();
+        if (auto baseInterfaceDeclRef = baseDeclRef->declRef.As<InterfaceDecl>())
+        {
+            for (auto subInheritanceDeclRef : getMembersOfType<InheritanceDecl>(baseInterfaceDeclRef))
+            {
+                auto cpyMangledName = getMangledNameForConformanceWitness(subType, subInheritanceDeclRef.getDecl()->getSup().type);
+                if (!witnessTablesDictionary.ContainsKey(cpyMangledName))
+                {
+                    auto cpyTable = context->irBuilder->createWitnessTable();
+                    cpyTable->mangledName = cpyMangledName;
+                    context->irBuilder->createWitnessTableEntry(witnessTable,
+                        context->irBuilder->getDeclRefVal(subInheritanceDeclRef), cpyTable);
+                    cpyTable->entries = witnessTable->entries;
```

Comment:
I don't know who currently owns the witness table entries, so this shallow copy of linked list pointers may be inappropriate.

Reply:
The intended ownership model for the IR is that all nodes for an IRModule should come out of a single pool, and we just drop that pool when destroying the module. None of that has been implemented, so everything leaks for now.

The biggest challenge in any of that is IR modes which contain pointers to reference-counted stuff; those need to be retained via a list of references stored on the module, I suppose.